src/main/java/hudson/plugins/git/browser/GitRepositoryBrowser.java

```diff
@@ -73,7 +74,7 @@ public final URL getUrl() throws IOException {
      *      null if the browser doesn't have any suitable URL.
      * @throws IOException on input or output error
      */
-    public abstract URL getFileLink(GitChangeSet.Path path) throws IOException;
+    public abstract URL getFileLink(GitChangeSet.Path path) throws IOException, URISyntaxException;
```

Comment:
I need advice from others on the relative safety of changing a public interface definition by throwing an additional type of exception.  Could you seek advice from at least one other person of your choice to advise if this is generally safe?

My personal preference would be to wrap the URISyntaxException inside an IOException and throw the IOException so that the method callers do not need to adapt to a new exception type.

Reply:
FWIW: Changing the throws clause does not affect binary compatibility (https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.21), but it would affect source compatibility.